File: LeetCode/Study Plan/Practice Problems/Medium/Find Minimum In Rotated Sorted Array/Go/find_minimum_in_rotated_sorted_array.go
package main

import "fmt"

func find_minimum_in_rotated_sorted_array(nums []int) int {
	res := nums[0]
	nums = nums[1:]
	l, r := 0, len(nums)-1
	for l <= r {
		mid := (l + r) / 2
		if res > nums[mid] {
			res = nums[mid]
			r = mid - 1
		}
		if res < nums[mid] {
			l = mid + 1
		}
	}
	return res
}

func main() {
	fmt.Println(find_minimum_in_rotated_sorted_array([]int{3, 4, 5, 1, 2}))
}

File: LeetCode/Study Plan/Practice Problems/Medium/Find Minimum In Rotated Sorted Array/Java/findMinimumInRotatedSortedArray.java
class Solution {
    public int findMin(int[] nums) {
        int low = 0, high = nums.length-1;
        while(low < high) {
            int mid = low + (high-low)/2;
            if (nums[low] <= nums[mid]) {
                if (nums[low] > nums[high]) {
                    low = mid+1;
                }else {
                    high = mid;
                }
            }else {
                if (nums[low] > nums[high]) {
                    high = mid;
                }else {
                    low = mid;
                }
            }
        }
        return nums[low];
    }

